robot/packages/cloud_package_manager.go

```diff
@@ -42,6 +43,8 @@ type managedPackage struct {
 	modtime    time.Time
 }
 
+var placeholderRegexp = regexp.MustCompile(`^\$\{(packages(\.(ml_models|modules))?\.[\w_\-]+)\}`)
```

Comment:
```suggestion
var placeholderRegexp = regexp.MustCompile(`\$\{(packages(\.(ml_models|modules))?\.[\w-]+)\}`)
```
Underscore is already included in \w and dash doesn't need to be escaped if it's the last character in a character set (square brackets) which it is here. Also, I don't _think_ you want to limit this to the start of the line with ^ either.

If I'm somehow wrong on any of those, let me know.

Reply:
Ah thanks for dash and underscore comment! I wasn't aware. 

I think we do now want to start this with a beginning of a string because we are no longer reading entire configs at a time but single strings at a time. So if this does not start with a $, then it is not a valid placeholder.

Reply:
Okay, as long as that single string is never going to be something like `python3 ${package.modules.myPyModule}` or something.

Reply:
@michaellee1019 do you have any insight to this? will placeholders for exec paths just be used to specified where the executable is? or will they ever be used as commands?